src/Thermochimica-cxx.C

```diff
@@ -0,0 +1,302 @@
+#include <stdio.h>
+#include <math.h>
+#include <cstring>
+#include <map>
+#include <vector>
+#include "Thermochimica.h"
+
+namespace Thermochimica
+{
+
+  void thermochimica()
+  {
+    TCAPI_thermochimica();
+  }
+
+  void setThermoFilename(const std::string &filename)
+  {
+    TCAPI_setThermoFilename(filename.c_str(), filename.length());
+  }
+
+  void setUnitTemperature(const std::string &tunit)
+  {
+    TCAPI_setUnitTemperature(tunit.c_str(), tunit.length());
+  }
+
+  void setUnitPressure(const std::string &punit)
+  {
+    TCAPI_setUnitPressure(punit.c_str(), punit.length());
+  }
+
+  void setUnitMass(const std::string &munit)
+  {
+    TCAPI_setUnitMass(munit.c_str(), munit.length());
+  }
+
+  void setUnits(const std::string &tunit, const std::string &punit, const std::string &munit)
+  {
+    setUnitTemperature(tunit);
+    setUnitPressure(punit);
+    setUnitMass(munit);
+  }
+
+  void setStandardUnits()
+  {
+    TCAPI_setStandardUnits();
+  }
+
+  void setModelicaUnits()
+  {
+    TCAPI_setModelicaUnits();
+  }
+
+  void setTemperaturePressure(double temperature, double pressure)
+  {
+    TCAPI_setTemperaturePressure(&temperature, &pressure);
+  }
+
+  void setElementMass(int element, double mass)
+  {
+    TCAPI_setElementMass(&element, &mass);
+  }
+
+  int checkInfoThermo()
+  {
+    int idbg;
+    TCAPI_checkInfoThermo(&idbg);
+    return idbg;
+  }
+
+  void parseThermoFile()
+  {
+    TCAPI_sSParseCSDataFile();
+  }
+
+  void setPrintResultsMode(int mode)
+  {
+    TCAPI_setPrintResultsMode(&mode);
+  }
+
+  void printResults()
+  {
+    TCAPI_printResults();
+  }
+
+  void resetInfoThermo()
+  {
+    TCAPI_resetInfoThermo();
+  }
+
+  void resetThermo()
+  {
+    TCAPI_resetThermo();
+  }
+
+  void resetThermoAll()
+  {
+    TCAPI_resetThermoAll();
+  }
+
+  // re-initialization-related functions
+  void saveReinitData()
+  {
+    TCAPI_saveReinitData();
+  }
+
+  std::pair<int, int> getReinitDataSizes()
+  {
+    int elements, species;
+    TCAPI_getReinitDataSizes(&elements, &species);
+    return {elements, species};
+  }
+
+  double getMolesPhase()
+  {
+    double molesPhase;
+    TCAPI_getMolesPhase(&molesPhase);
+    return molesPhase;
+  }
```

Comment:
```suggestion
  syd::vector<double> getMolesPhase()
  {
    std::vector<double> molesPhase(nElements);
    TCAPI_getMolesPhase(molesPhase.data());
    return molesPhase;
  }
```

This routine must return a vector rather than a single number. The length of the vector is the number of elements in the system. We'll need to call an API (to be added) function here to get `nElements`.

Reply:
I pushed a fix for this, but didn't use this suggestion exactly. Thanks!